src/Microsoft.TemplateEngine.Cli/HelpForTemplateResolution.cs

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Edge.Template;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli
+{
+    internal static class HelpForTemplateResolution
+    {
+        public static void DisplayInvalidParameters(IReadOnlyList<string> invalidParams)
+        {
+            if (invalidParams.Count > 0)
+            {
+                Reporter.Error.WriteLine(LocalizableStrings.InvalidInputSwitch.Bold().Red());
+                foreach (string flag in invalidParams)
+                {
+                    Reporter.Error.WriteLine($"  {flag}".Bold().Red());
+                }
+            }
+        }
+
+        public static void DisplayParametersInvalidForSomeTemplates(IReadOnlyList<string> invalidParams)
+        {
+            if (invalidParams.Count > 0)
+            {
+                Reporter.Error.WriteLine(LocalizableStrings.PartialTemplateMatchSwitchesNotValidForAllMatches.Bold().Red());
+                foreach (string flag in invalidParams)
+                {
+                    Reporter.Error.WriteLine($"  {flag}".Bold().Red());
+                }
+            }
+        }
+
+        // get a better name for this, it's unclear what it's doing.
```

Comment:
Is this something we want to address now or in the future? I'm OK either way - if it's for the future, let's prefix with `TODO :` so it's easier to find in the future

Reply:
It should be a "TODO:", I'll add it.